process/coordinator/process.go

```diff
@@ -693,6 +706,7 @@ func (tc *transactionCoordinator) processCompleteMiniBlock(
 			log.Error(errAccountState.Error())
 		}
 
+		tc.gasHandler.SetGasConsumed(currentGasConsumedByBlock)
```

Comment:
why is this needed ?

Reply:
We need to revert to the snapshot of currentGasConsumedByBlock which is saved above, exactly as we do for account state, if in ProcessMiniBlock which processes each tx one by one changing the consumed gas, there is an error and we should drop this miniblock because  it can not be fully processed